src/components/settings/SettingsLMSTab/ErrorReporting/utils.jsx

```diff
@@ -8,6 +8,11 @@ import {
 
 timeago.register('time-locale');
 
+export function createLookup(source, lookupFunc, transformation) {
+  const newFiltersMap = source.map((item) => ({ [lookupFunc(item)]: transformation(item) }));
+  return Object.assign({}, ...newFiltersMap);
+}
```

Comment:
nit: Should document how this works.  It's a common piece of functionality, but no library I know of has an exact match so people won't know immediately what it's doing.